src/common/PlatformInformation.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import { executeCommandSync } from './cpUtils';
+
+export class PlatformInformation {
+    public constructor(private platform: string, private architecture: string, public rid: string) { }
+
+    public isWindows(): boolean {
+        return this.platform === 'win32';
+    }
+
+    public isMacOS(): boolean {
+        return this.platform === 'darwin';
+    }
+
+    public toString(): string {
+        let result = `${this.rid} (${this.platform}`;
+
+        if (this.architecture) {
+            if (result) {
+                result += ', ';
+            }
+
+            result += this.architecture;
+        }
+        result += ')';
+
+        return result;
+    }
+
+    public static async GetCurrent(): Promise<PlatformInformation> {
+        let platform = os.platform();
+        let architecture: string;
+        let rid: string;
+
+        switch (platform) {
+            case 'win32':
+                architecture = await PlatformInformation.GetWindowsArchitecture();
+                rid = 'win-';
+                break;
+
+            case 'darwin':
+                architecture = await PlatformInformation.GetUnixArchitecture();
+                rid = 'osx-';
+                break;
+
+            default:
+                throw new Error(`Unsupported platform: ${platform}`);
+        }
+        if (architecture === 'x86') {
```

Comment:
we currently only support x64, use a whitelist in case it is running on ARM
```suggestion
        if (architecture !== 'x64') {
``` #Resolved